database/functions.py

```
@@ -27,3 +29,33 @@ def add_hero_with_his_powers(new_hero: NewHero) -> dict[str, Any]:
     powers = hero_dict.pop('powers')
     hero = add_hero_with_powers(hero_dict, powers)
     return HeroSchema.from_orm(hero).model_dump(mode='json')
+
+
+def set_hero_is_retired(hero_id: int) -> Optional[Hero]:
+    """
+    If the hero exists, is_retired field sets to True.
+
+    Query Parameters:
+    - hero_id: int
+    """
+    hero = get_hero_by_id(hero_id=int(hero_id))
+    if not hero:
+        return {'error': 'Hero not found'}
```

I think if we can simplify this by avoiding the check inside the function itself. for example we could raise a NoResultFound exception in get_hero_by_id when the hero doesn't exist, and then handle it in the commit_db decorator. this might help keep the function logic cleaner.